--- origsrc/demos/src/egl/oes_vg/Makefile.am	2012-05-30 02:25:18.000000000 -0500
+++ src/demos/src/egl/oes_vg/Makefile.am	2012-07-06 05:08:01.047348100 -0500
@@ -23,13 +23,13 @@
 #    Cooper Yuan <cooperyuan@gmail.com>
 
 AM_CFLAGS = \
-	$(GLESV1_CFLAGS) \
+	$(GL_CFLAGS) \
 	$(EGL_CFLAGS) \
 	$(VG_CFLAGS) \
 	$(X11_CFLAGS)
 
 AM_LDFLAGS = \
-	$(GLESV1_LIBS) \
+	$(GL_LIBS) \
 	$(EGL_LIBS) \
 	$(VG_LIBS) \
 	-lm
