--- origsrc/mesa-demos-8.2.0/src/egl/opengl/Makefile.am	2014-07-05 03:17:34.000000000 -0500
+++ src/mesa-demos-8.2.0/src/egl/opengl/Makefile.am	2015-05-01 13:09:59.241328700 -0500
@@ -50,16 +50,18 @@ endif
 endif
 
 if HAVE_EGL
-noinst_PROGRAMS = \
-	demo1 \
-	demo2 \
-	demo3 \
+bin_PROGRAMS = \
 	eglinfo \
 	eglgears_screen \
 	egltri_screen \
 	peglgears \
 	$(EGL_DRM_DEMOS) \
 	$(EGL_X11_DEMOS)
+
+noinst_PROGRAMS = \
+	demo1 \
+	demo2 \
+	demo3
 endif
 
 egltri_screen_SOURCES = egltri.c
